Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -1113,42 +1115,29 @@ def fetch_incidents():
     last_run = demisto.getLastRun()
     # Get the last fetch time, if exists
     last_fetch = last_run.get('first_behavior_time')
+    offset = last_run.get('offset', 0)
 
     # Handle first time fetch, fetch incidents retroactively
     if last_fetch is None:
         last_fetch, _ = parse_date_range(FETCH_TIME, date_format='%Y-%m-%dT%H:%M:%SZ')
+    prev_fetch = last_fetch
 
     last_fetch_timestamp = int(parse(last_fetch).timestamp() * 1000)
 
-    last_detection_id = str(last_run.get('last_detection_id'))
+    # last_detection_id = str(last_run.get('last_detection_id'))
 
     fetch_query = demisto.params().get('fetch_query')
 
     if fetch_query:
         fetch_query = "created_timestamp:>'{time}'+{query}".format(time=last_fetch, query=fetch_query)
-        detections_ids = demisto.get(get_fetch_detections(filter_arg=fetch_query), 'resources')
+        detections_ids = demisto.get(get_fetch_detections(filter_arg=fetch_query, offset=offset), 'resources')
 
     else:
-        detections_ids = demisto.get(get_fetch_detections(last_created_timestamp=last_fetch), 'resources')
+        detections_ids = demisto.get(get_fetch_detections(last_created_timestamp=last_fetch, offset=offset),
+                                     'resources')
     incidents = []  # type:List
 
     if detections_ids:
-
-        # make sure we do not fetch the same detection again.
-        if last_detection_id == detections_ids[0]:
-            first_index_to_fetch = 1
-
-            # if this is the only detection - dont fetch.
-            if len(detections_ids) == 1:
-                return incidents
-
-        # if the first detection in this pull is different than the last detection fetched we bring it as well
-        else:
-            first_index_to_fetch = 0
-
-        # Limit the results to INCIDENTS_PER_FETCH`z
-        last_index_to_fetch = INCIDENTS_PER_FETCH + first_index_to_fetch
-        detections_ids = detections_ids[first_index_to_fetch:last_index_to_fetch]
```

Comment:
restore these lines.
no need to iterate and create an incident object for detections you are not going to return.

Reply:
as I'm using offset I always get the number of detections I want (or less) so this is no longer needed.